src/main/scala/chisel3/util/TransitName.scala

```diff
@@ -2,25 +2,129 @@
 
 package chisel3.util
 
-import chisel3._
-import internal.HasId
+import chisel3.internal.HasId
 
-/**
- * The purpose of TransitName is to allow a library to 'move' a name
- * call to a more appropriate place.
- * For example, a library factory function may create a module and return
- * the io. The only user-exposed field is that given IO, which can't use
- * any name supplied by the user. This can add a hook so that the supplied
- * name then names the Module.
- * See Queue companion object for working example
- */
+/** The purpose of [[TransitName]] is to allow a library writer to "move" a name call to a more appropriate place. The
+  * result of this is that you can get better names for generated hardware.
+  *
+  * Consider the example below. This contains a simple module called `Not` that implements the functionality of an
+  * [[https://en.wikipedia.org/wiki/Inverter_(logic_gate) inverter]]. For convenience, this has a companion object that
+  * implements the [[https://en.wikipedia.org/wiki/Factory_method_pattern Factory Method Pattern]] to create instances
+  * of the module `Not`. However, the factory only returns the IO of the instantiated module.
+  *
+  * By using [[TransitName]] you can "transit" the name ''from'' the IO (that the user specifies with a `val`) ''to''
+  * the instance. This results in better names in the resulting FIRRTL or Verilog.
+  *
+  * {{{
```

Comment:
Should this example be duplicated or could we just point people to the testcase here?

Reply:
Let me simplify this somewhat... Adding the do not merge tag for now.

Reply:
Simplifying this to only refer to a portion of the testcase.